compiler/common-artifacts/src/TestDataGenerator.cpp

```diff
@@ -62,10 +63,12 @@ template <typename T> void geneate_random_data(std::mt19937 &gen, void *data, ui
   }
 }
 
-void fill_random_data(void *data, uint32_t size, loco::DataType dtype)
+void fill_random_data(void *data, uint32_t size, loco::DataType dtype, uint32_t seed = 0)
```

Comment:
we usually didn't use default values for methods. can you introduce new `fill_random_data` with 4 parameters and then let `fill_random_data` with 3 call with 4 arguments for intermediate state and then gradually remove with 3 ?